parcels/kernel.py

```diff
@@ -411,11 +405,13 @@ def remove_deleted(pset):
         remove_deleted(pset)
 
         # Identify particles that threw errors
-        error_particles = np.isin(pset.particle_data['state'], [StateCode.Success, StateCode.Evaluate], invert=True)
+        # NOTE: this relies on the SOA implementation, it does not
+        # generalize. This needs to be reworked.
+        error_particles = np.isin(pset.collection._data['state'], [StateCode.Success, StateCode.Evaluate], invert=True)
```

Comment:
okay, this should rather be a list comprehension - just as it was prior to SoA and as it is in all branches. Reasoning here is that this execute function should later be a generic super-class function, where only the `execute_[ji|python]` actually change.

Can be in here for now as-is, but it will be subject to one of the earliest follow-up changes.